src/chainstate/burn/mod.rs

```diff
@@ -179,12 +180,22 @@ impl SortitionHash {
         SortitionHash(ret)
     }
 
-    /// Choose 1 index from the range [0, max).
-    pub fn choose(&self, max: u32) -> u32 {
+    /// Choose two indices (without replacement) from the range [0, max).
+    pub fn choose_two(&self, max: u32) -> Vec<u32> {
```

Comment:
Not required for this PR, but just in case the parameter choice somehow changes _again_, do you think you could make this method more general (e.g. `choose_n(&self, max: u32, n: u32) -> Vec<u32>`)?

Reply:
It's possible, but I'd prefer we didn't until we wanted to change the parameter choice. The reason for that is that the way we implement choosing `n` from `max` is consensus bearing, and a generalized approach to do this increases the complexity here when we probably don't want it to (we'd need to do a general swap-and-shrink on the possible indexes).